### ISSUE TITLE
Bootstrap tour conflicts with other Bootstrap Modals

### ISSUE THREAD

**Issue by PsychoIndigo** [sorich87/bootstrap-tour#718]
I don't know if it's because of the z-index, but I have two modals on one page that don't work because I installed bootstrap tour on the same page. Only the bootstrap tour works and the other modals only show the black backdrop.

**Comment by IGreatlyDislikeJavascript** [comment]
Can you explain the problem in more detail - are the modals part of a tour, are all tour steps inside a modal, is the tour unrelated to the modal but it's causing visual issues etc? Are you using modals precreated in the dom or are you using a plugin like bootstrap modal to programmatically generate them?

There are a number of issues getting modals to work with tour because of the way bootstrap modals are created vs the way bootstrap tour expects dom elements.

Try my fork and see if that fixes it.

**Comment by redelement** [comment]
It is not Bootstrap 4 compatible. If you use the standalone version it will function but modals will not function properly as the .fade classes in the stylesheet prevent the modals from opening properly. Simply removing them does not solve the problem either as there are a few incompatible points in the JS as well. I have just encountered this problem and will reply with a fix when I have one.

**Comment by IGreatlyDislikeJavascript** [comment]
@redelement please can you check if my fork of Tour fixes this for you.
https://github.com/IGreatlyDislikeJavascript/bootstrap-tourist

I've specifically added robust modal handling as well:
https://github.com/IGreatlyDislikeJavascript/bootstrap-tourist#trigger-when-modal-closes

**Comment by redelement** [comment]
@IGreatlyDislikeJavascript it does solve my original problems, however I ran into more theme specific problems that just make this incompatible with my project so I ended up using a different plugin. For anybody using vanilla Bootstrap 4 I believe your fork will solve the issue.

**Comment by ckgagan** [comment]
Yeah I faced the same problem. It is because of the fade class. Bootstrap modal adds fade class in its modal element.
Bootstrap tour has a css property where fade class has opacity of 0, so we do not see the bootstrap modal but just the backdrop.
If we remove the fade class from Bootstrap modal, then it should work fine.

**Comment by PsychoIndigo** [comment]
Damn! Thank you! 🖤🧡🖤

Sent from Yahoo Mail on Android 
 
  On Tue, Oct 6, 2020 at 10:38 PM, Gagan Shrestha<notifications@github.com> wrote:   


Yeah I faced the same problem. It is because of the fade class. Bootstrap modal adds fade class in its modal element.
Bootstrap tour has a css property where fade class has opacity of 0, so we do not see the bootstrap modal but just the backdrop.
If we remove the fade class from Bootstrap modal, then it should work fine.

**Comment by alanish08** [comment]
> 
> 
> Yeah I faced the same problem. It is because of the fade class. Bootstrap modal adds fade class in its modal element.
> Bootstrap tour has a css property where fade class has opacity of 0, so we do not see the bootstrap modal but just the backdrop.
> If we remove the fade class from Bootstrap modal, then it should work fine.
> OR the tour plugin must use another class instead of **fade**.

Thanks man

**Comment by super1zero** [comment]
One solution that might work is to use specific stylesheet to specific part of html i.e for bootstrap-tour use bootstrap-tour stylesheet and leave rest as it is:
https://stackoverflow.com/questions/45572399/applying-a-stylesheet-to-only-a-certain-region-of-an-html-file
